pkg/worker/redis.go

```diff
@@ -188,29 +189,31 @@ func (w *RedisWorker) fetchJob(ctx context.Context) {
 	var job Job
 	dec := gob.NewDecoder(strings.NewReader(res[1]))
 	err = dec.Decode(&job)
-	logger := ptr.To(zerolog.Ctx(ctx).With().
-		Str("job_id", job.ID.String()).
-		Str("job_type", job.Type.String()).
-		Interface("job_args", job.Args).
-		Logger())
 	if err != nil {
-		logger.Error().Err(err).Msg("Unable to unmarshal job payload, skipping")
+		zerolog.Ctx(ctx).Error().
+			Err(err).
+			Str("job_id", job.ID.String()).
+			Str("job_type", job.Type.String()).
+			Interface("job_args", job.Args).
+			Msg("Unable to unmarshal job payload, skipping")
+		return
 	}
 
 	atomic.AddInt64(&w.inFlight, 1)
+	defer atomic.AddInt64(&w.inFlight, -1)
+
 	w.processJob(ctx, &job)
 }
 
 func (w *RedisWorker) processJob(origCtx context.Context, job *Job) {
 	if job == nil {
 		return
 	}
-	defer atomic.AddInt64(&w.inFlight, -1)
```

Comment:
Why are you moving this? I am fine, just curious.

Reply:
I've moved it to `fetchJob` so it's together with the +1 ... if felt bit disjoined.
And if I understand it correctly, it should work the same, right?